tests/iamserver/iamserver_test.cpp

```diff
@@ -252,739 +255,200 @@ void IAMServerTest::ApplyCertificate(const String& certType, const String& subje
 
-    const auto status = clientStub->GetSystemInfo(&context, request, &response);
-    ASSERT_TRUE(status.ok()) << status.error_message() << " (" << status.error_code() << ")";
-    ASSERT_EQ(response.system_id(), cSystemID);
-    ASSERT_EQ(response.unit_model(), cUnitModel);
-}
+    auto err = mServer.Init(mServerConfig, mCertHandler, mIdentHandler, mPermHandler, mCertLoader, mCryptoProvider,
+        mNodeInfoProvider, mNodeManager, mProvisionManager, cProvisioningModeOn);
 
-TEST_F(IAMServerTest, GetSystemInfoFailsOnSystemId)
-{
-    auto err = mServer.Init(mServerConfig, mCertHandler, &mIdentHandler, &mPermHandler, mRemoteIAMHandler.get(),
-        mCertLoader, mCryptoProvider, mNodeInfoProvider, cProvisioningModeOff);
     ASSERT_TRUE(err.IsNone()) << err.Message();
 
-    auto clientStub = CreateCustomStub<IAMPublicIdentityService>(mClientInfo, mServerConfig.mIAMProtectedServerURL);
-    ASSERT_NE(clientStub, nullptr) << "Failed to create client stub";
+    auto stub = CreateCustomStub<iamproto::IAMPublicIdentityService>(
+        mServerConfig.mIAMProtectedServerURL, cProvisioningModeOn);
 
-    grpc::ClientContext     context;
-    google::protobuf::Empty request;
-    SystemInfo              response;
+    EXPECT_NE(stub, nullptr) << "Failed to create a stub";
 
-    EXPECT_CALL(mIdentHandler, GetSystemID)
-        .WillOnce(Return(aos::RetWithError<aos::StaticString<aos::cSystemIDLen>>("", aos::ErrorEnum::eFailed)));
-    EXPECT_CALL(mIdentHandler, GetUnitModel).Times(0);
+    grpc::ClientContext  context;
+    iamproto::SystemInfo response;
 
-    const auto status = clientStub->GetSystemInfo(&context, request, &response);
-    ASSERT_FALSE(status.ok()) << status.error_message() << " (" << status.error_code() << ")";
-}
+    auto status = stub->GetSystemInfo(&context, {}, &response);
 
-TEST_F(IAMServerTest, GetSystemInfoFailsOnUnitModel)
-{
-    auto err = mServer.Init(mServerConfig, mCertHandler, &mIdentHandler, &mPermHandler, mRemoteIAMHandler.get(),
-        mCertLoader, mCryptoProvider, mNodeInfoProvider, cProvisioningModeOff);
-    ASSERT_TRUE(err.IsNone()) << err.Message();
-
-    auto clientStub = CreateCustomStub<IAMPublicIdentityService>(mClientInfo, mServerConfig.mIAMProtectedServerURL);
-    ASSERT_NE(clientStub, nullptr) << "Failed to create client stub";
-
-    grpc::ClientContext     context;
-    google::protobuf::Empty request;
-    SystemInfo              response;
-
-    EXPECT_CALL(mIdentHandler, GetSystemID)
-        .WillOnce(Return(aos::RetWithError<aos::StaticString<aos::cSystemIDLen>>(cSystemID)));
-    EXPECT_CALL(mIdentHandler, GetUnitModel)
-        .WillOnce(Return(aos::RetWithError<aos::StaticString<aos::cUnitModelLen>>("", aos::ErrorEnum::eFailed)));
-
-    const auto status = clientStub->GetSystemInfo(&context, request, &response);
-    ASSERT_FALSE(status.ok()) << status.error_message() << " (" << status.error_code() << ")";
+    EXPECT_EQ(status.error_code(), grpc::StatusCode::UNIMPLEMENTED)
```

Comment:
I think iamserver_test.cpp should be removed, anyway test server doesn't implement anything here

Reply:
I'd keep it
This UT checks iam server init
Provision flag driven logic
and availability of IAM public services on the protected port

Reply:
so, basically it checks Init method only.
So I would suggest keep unit tests for Init method only and remove all the rest.